docs/content/docs/pub-sub-implementations.md

```diff
@@ -162,21 +160,34 @@ When implemented, HTTP publisher can be used as webhooks sender.
 | GuaranteedOrder | yes |  |
 | Persistent | no| |
 
-#### Configuration
+#### Subscriber configuration
 
-The configuration of HTTP subscriber is done via the constructor.
+Subscriber configuration is done via the config struct passed to the constructor:
 
 {{% render-md %}}
-{{% load-snippet-partial file="content/src-link/message/infrastructure/http/subscriber.go" first_line_contains="// NewSubscriber" last_line_contains="func NewSubscriber(" %}}
+{{% load-snippet-partial file="content/src-link/message/infrastructure/http/subscriber.go" first_line_contains="type SubscriberConfig struct" last_line_contains="}" %}}
 {{% /render-md %}}
 
-You can also use `NewSubscriberWithRouter` constructor to pass your own `chi.Router` (see [chi](https://github.com/go-chi/chi)).
+You can use the `Router` config option to `SubscriberConfig` to pass your own `chi.Router` (see [chi](https://github.com/go-chi/chi)).
 This may be helpful if you'd like to add your own HTTP handlers (e.g. a health check endpoint).
 
+#### Publisher configuration
+
+Publisher configuration is done via the config struct passed to the constructor:
+
 {{% render-md %}}
-{{% load-snippet-partial file="content/src-link/message/infrastructure/http/subscriber.go" first_line_contains="// NewSubscriberWithRouter" last_line_contains="(*Subscriber, error)" %}}
+{{% load-snippet-partial file="content/src-link/message/infrastructure/http/publisher.go" first_line_contains="type PublisherConfig struct" last_line_contains="}" %}}
 {{% /render-md %}}
 
+How the message topic and body translate into the URL, method and payload of the HTTP request is highly configurable through the use of `MarshalMessageFunc`. 
+Use the provided `DefaultMarshalMessageFunc` to send POST requests to a predefined address:
```

Comment:
some tip that you probably want to implement your own marshal function will be useful

Reply:
umm that's quite obvious, I guess